kie-server-parent/kie-server-services/kie-server-services-prometheus/src/main/java/org/kie/server/services/prometheus/PrometheusJobListener.java

```diff
@@ -87,6 +101,12 @@ public void afterJobExecuted(AsynchronousJobEvent event) {
         final RequestInfo job = event.getJob();
         numberOfJobsExecuted.labels(defaultString(job.getDeploymentId()), String.valueOf(event.failed()), job.getCommandName()).inc();
         numberOfRunningJobs.labels(defaultString(job.getDeploymentId()), job.getCommandName()).dec();
+        if(job.getRetries() > 0 && job.getStatus().equals(STATUS.RETRYING)) {
+        	numberOfJobsRetrying.labels(defaultString(job.getDeploymentId()), String.valueOf(event.failed()), job.getCommandName()).inc();
+        }
+        else if(job.getRetries() == 0 && job.getStatus().equals(STATUS.ERROR)) {        	
+        	numberOfJobsErrored.labels(defaultString(job.getDeploymentId()), String.valueOf(event.failed()), job.getCommandName()).inc();        	
```

Comment:
probably not the best approach. You need to do this afterJobExecution not here during scheduling.
Also when  job.getRetries() == 0 numberOfJobsRetrying should be decreased